### PR TITLE
Nv 3006 - the new chat editor

### PR DESCRIPTION
**Pull request by ainouzgali**
### What change does this PR introduce?

Create the new Chat editor with the preview 

Screen Shot 2024-02-12 at 14 47 36
Screen Shot 2024-02-12 at 14 47 45

### CODE REVIEW

File: apps/web/src/pages/templates/components/chat-editor/TemplateChatEditor.tsx [hunk 1]
```diff
@@ -1,51 +1,76 @@
...
-        )}
+        </Grid.Col>
+        <Grid.Col span={6}>
+          <ChatPreview showLoading={showLoading} />
+        </Grid.Col>
+      </Grid>
+      <VariableManagerModal
+        open={variablesModalOpened}
+        setOpen={setVariablesModalOpen}
+        variablesArray={variablesArray}
```
Comment (BiswaViraj) [inline apps/web/src/pages/templates/components/chat-editor/TemplateChatEditor.tsx:72]: Can we move this to the `VariableManagementButton` as both are inter-dependent design wise.

File: apps/web/src/pages/templates/components/chat-editor/TemplateChatEditor.tsx [hunk 2]
```diff
@@ -1,51 +1,76 @@
...
   });
+  const [variablesModalOpened, setVariablesModalOpen] = useState(false);
+
+  const { start, clear } = useTimeout(() => setShowLoading(false), 1000);
+  const handleContentChange = (value: string, onChange: (string) => void) => {
+    setShowLoading(true);
+    clear();
+    onChange(value);
+    start();
+  };
```
Comment (BiswaViraj) [inline apps/web/src/pages/templates/components/chat-editor/TemplateChatEditor.tsx:36]: Suggestion: let's add a comment here why we need the onChange in a timeOut

Comment (BiswaViraj) [reply apps/web/src/pages/templates/components/chat-editor/TemplateChatEditor.tsx:36]: Also, i think we will be using the same logic for other channels too, might be better to abstract this behaviour into a hook. wdyt?

**Review by BiswaViraj — COMMENTED** [review]
Awesome work 🚀 
Left a few small suggestions

File: apps/web/src/pages/templates/components/chat-editor/TemplateChatEditor.tsx (hunk 2 above)
Comment (ainouzgali) [reply apps/web/src/pages/templates/components/chat-editor/TemplateChatEditor.tsx:36]: Yes, I thought the same, it will be useful. I'll do it

**Review by BiswaViraj — APPROVED** [review]
🚀

**Pull request merged**